app/components/CreateCrashRecordModal.tsx

```diff
@@ -0,0 +1,377 @@
+import Button from "react-bootstrap/Button";
+import Card from "react-bootstrap/Card";
+import Form from "react-bootstrap/Form";
+import Modal from "react-bootstrap/Modal";
+import Spinner from "react-bootstrap/Spinner";
+import AlignedLabel from "@/components/AlignedLabel";
+import FormControlDatePicker from "@/components/FormControlDatePicker";
+import { FaCirclePlus, FaCircleMinus } from "react-icons/fa6";
+import { useForm, SubmitHandler, useFieldArray } from "react-hook-form";
+import { useQuery, useMutation } from "@/utils/graphql";
+import { UNIT_TYPES_QUERY } from "@/queries/unit";
+import { CREATE_CRASH } from "@/queries/crash";
+import { LookupTableOption } from "@/types/relationships";
+import { Crash } from "@/types/crashes";
+import { useAuth0 } from "@auth0/auth0-react";
+
+interface CreateCrashModalProps {
+  /**
+   * A callback fired when either the modal backdrop is clicked, or the
+   * escape key is pressed
+   */
+  onClose: () => void;
+  /**
+   *  Callback fired after the user form is successfully submitted
+   */
+  onSubmitCallback: () => void;
+  /**
+   * If the modal should be visible or hidden
+   */
+  show: boolean;
+}
+
+type UnitInputs = {
+  unit_desc_id: string;
+  fatality_count: number;
+  injury_count: number;
+};
+
+type CrashInputs = {
```

Comment:
so nice to have all the expected fields for a temp crash in this type and the defaults below ✨